test/libsolidity/smtCheckerTests/special/msg_value_inheritance_2.sol

```diff
@@ -0,0 +1,18 @@
+pragma experimental SMTChecker;
+
+contract A {
+	uint public x = msg.value;
+	constructor() {
+		assert(x == 0); // should fail, if A is constructed as part of C, it can have any msg.value
+	}
+}
+
+contract C is A {
+	uint public v = msg.value; // 1
+	constructor() A() payable {
+		assert(v == 0); // should fail, C can be constructed with any msg.value
+	}
+}
+// ----
+// Warning 6328: (93-107): CHC: Assertion violation happens here.\nCounterexample:\nv = 0, x = 1\n\nTransaction trace:\nC.constructor(){ value: 1 }
+// Warning 6328: (273-287): CHC: Assertion violation happens here.\nCounterexample:\nv = 1, x = 1\n\nTransaction trace:\nC.constructor(){ value: 1 }
```

Comment:
Nice! Wdyt about following test configuration?

contract A: non payable
contract B: non payable
contract C -> A, B: payable

I assume `msg.value == 0` cannot hold in any of the constructors, right?

Another test config

contract A: payable
contract B -> A: non payable
contract C -> B, A: non payable

Reply:
Yes, the situation should be the same as in the two-contract inheritance covered by `msg_value_inheritance_1.sol` and `msg_value_inheritance_2.sol`. Not sure if the tests would bring additional value, but I can add them if you believe so.

Reply:
I'm not sure if it will bring value to be honest, was just thinking out loud :)

```
contract A {
  uint x = msg.value;
  constructor() {
    assert(x == 0);
  }
}
contract B {
  constructor() payable {
    assert(msg.value >= 0);
  }
}
contract C is A, B {
  constructor() A() B() payable {
    assert(msg.value >= 0);
  }
}
```

I was wondering which of the assertions above are thrown

Reply:
Only the first one is reported, it can be violated when `A` is constructed  as part of `C`. I am adding this test case.